tests/performance_tests/performance_test_basic.py

```diff
@@ -14,33 +14,66 @@
 from Node import Node
 from Node import ReturnType
 from TestHelper import TestHelper
+from dataclasses import dataclass
 
 Print = Utils.Print
 errorExit = Utils.errorExit
 cmdError = Utils.cmdError
 relaunchTimeout = 30
 emptyBlockGoal = 5
 
-def waitForEmptyBlocks(node1):
+@dataclass
+class blockData():
+    partialBlockId: str = ""
+    blockNum: int = 0
+    transactions: int = 0
+    net: int = 0
+    cpu: int = 0
+    elapsed: int = 0
+    time: int = 0
+    latency: int = 0
+
+class chainData():
+    def __init__(self):
+        self.blockLog = []
+        self.startBlock = 0
+        self.ceaseBlock = 0
+        self.totalTransactions = 0
+        self.totalNet = 0
+        self.totalCpu = 0
+        self.totalElapsed = 0
+        self.totalTime = 0
+        self.totalLatency = 0
+    def updateTotal(self, transactions, net, cpu, elapsed, time, latency):
+        self.totalTransactions += transactions
+        self.totalNet += net
+        self.totalCpu += cpu
+        self.totalElapsed += elapsed
+        self.totalTime += time
+        self.totalLatency += latency
+    def __str__(self):
+        return (f"Starting block: {self.startBlock}\nEnding block:{self.ceaseBlock}\nChain transactions: {self.totalTransactions}\n"
+         f"Chain cpu: {self.totalNet}\nChain net: {self.totalCpu}\nChain elapsed: {self.totalElapsed}\nChain time: {self.totalTime}\nChain latency: {self.totalLatency}")
+
+def waitForEmptyBlocks(node):
     emptyBlocks = 0
     while emptyBlocks < emptyBlockGoal:
-        headBlock = node1.getHeadBlockNum()
-        block = node1.processCurlCmd("chain", "get_block_info", f'{{"block_num":{headBlock}}}', silentErrors=False, exitOnError=True)
-        node1.waitForHeadToAdvance()
+        headBlock = node.getHeadBlockNum()
+        block = node.processCurlCmd("chain", "get_block_info", f'{{"block_num":{headBlock}}}', silentErrors=False, exitOnError=True)
+        node.waitForHeadToAdvance()
         if block['transaction_mroot'] == "0000000000000000000000000000000000000000000000000000000000000000":
             emptyBlocks += 1
         else:
             emptyBlocks = 0
+    return node.getHeadBlockNum()
 
-def checkTotalTrx():
-    total = 0
-    f = open("var/lib/node_00/stderr.txt")
-    stringResult = re.findall(r'trxs:\s+\d+', f.read())
-    for value in stringResult:
-        intResult = re.findall(r'\d+', value)
-        total += int(intResult[0])
-    f.close()
-    return total
+def fetchStats(total):
+    with open("var/lib/node_01/stderr.txt") as f:
+        blockResult = re.findall(r'Received block ([0-9a-fA-F]*).* #(\d+) .*trxs: (\d+),.*, net: (\d+), cpu: (\d+), elapsed: (\d+), time: (\d+), latency: (-?\d+) ms', f.read())
```

Comment:
This will not match 2.x.x or even 3.1.0 output; they do not contain `net`, or `cpu` or `elapsed` or `time`